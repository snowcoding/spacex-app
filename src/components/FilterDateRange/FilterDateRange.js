import React, { useContext } from 'react'
import { makeStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'
import { Slider } from '@material-ui/core'
import {
  LaunchContext,
  changeFilterDateRange,
  changePaginationPage,
} from '../../contexts/LaunchProvider'
import './filterDateRange.scss'

const useStyles = makeStyles({
  root: {
    width: 175,
    display: 'flex',
    flexDirection: 'column',
    marginLeft: 13,
    marginTop: 30,
    marginBottom: 30,
  },
})

function valuetext(value) {
  return `${value}`
}

const marks = [
  { value: 2006, label: '2006' },
  { value: 2007 },
  { value: 2008 },
  { value: 2009 },
  { value: 2010 },
  { value: 2011 },
  { value: 2012 },
  { value: 2013, label: '2013' },
  { value: 2014 },
  { value: 2015 },
  { value: 2016 },
  { value: 2017 },
  { value: 2018 },
  { value: 2019 },
  { value: 2020, label: '2020' },
]

export default function FilterDateRange() {
  const classes = useStyles()
  const [state, dispatch] = useContext(LaunchContext)

  const handleChange = (event, newValue) => {
    dispatch(changeFilterDateRange(newValue))
    if (state.paginationPage > 0) dispatch(changePaginationPage(0))
  }

  return (
    <div className={classes.root}>
      <Typography id='range-slider' gutterBottom className='date-range-label'>
        Date
      </Typography>
      <Slider
        value={state.filterDateRange}
        onChange={handleChange}
        valueLabelDisplay='auto'
        aria-labelledby='range-slider'
        getAriaValueText={valuetext}
        marks={marks}
        step={1}
        min={2006}
        max={2020}
      />
    </div>
  )
}
