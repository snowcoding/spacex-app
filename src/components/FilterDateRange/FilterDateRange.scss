@import '../../globals.scss';

.date-range-label {
    color: $mui-primary;
}