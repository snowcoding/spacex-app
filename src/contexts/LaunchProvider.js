import React, { createContext, useReducer } from 'react'

export const LaunchContext = createContext()

const reducer = (state, action) => {
  console.log('Launch Filter State: ', state)
  console.log('Launch Filter Action: ', action)
  switch (action.type) {
    case 'changeFilterDateRange':
      return {
        ...state,
        filterDateRange: [...action.filterDateRange],
      }

    default:
      throw new Error()
  }
}

export const changeFilterDateRange = filterDateRange => {
  return {
    type: 'changeFilterDateRange',
    filterDateRange,
  }
}

const LaunchProvider = props => {
  const initialState = {
    filterDateRange: [2006, 2020],
  }

  const [state, dispatch] = useReducer(reducer, initialState)

  return (
    <LaunchContext.Provider value={[state, dispatch]}>
      {props.children}
    </LaunchContext.Provider>
  )
}

export default LaunchProvider
