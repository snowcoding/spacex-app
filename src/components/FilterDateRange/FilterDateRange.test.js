import React from 'react'
import ReactDOM from 'react-dom'
import FilterDateRange from './FilterDateRange'
import LaunchProvider from '../../contexts/LaunchProvider'
import renderer from 'react-test-renderer'

it('snapshot test Date Range Filter', () => {
  const tree = renderer
    .create(
      <LaunchProvider>
        <FilterDateRange />
      </LaunchProvider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <LaunchProvider>
      <FilterDateRange />
    </LaunchProvider>,
    div
  )
  ReactDOM.unmountComponentAtNode(div)
})
