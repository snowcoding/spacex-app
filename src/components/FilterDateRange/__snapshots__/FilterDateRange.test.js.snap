// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test Date Range Filter 1`] = `
<div
  className="makeStyles-root-1"
>
  <p
    className="MuiTypography-root date-range-label MuiTypography-body1 MuiTypography-gutterBottom"
    id="range-slider"
  >
    Date
  </p>
  <span
    className="MuiSlider-root MuiSlider-marked"
    onMouseDown={[Function]}
  >
    <span
      className="MuiSlider-rail"
    />
    <span
      className="MuiSlider-track"
      style={
        Object {
          "left": "0%",
          "width": "100%",
        }
      }
    />
    <input
      type="hidden"
      value="2006,2020"
    />
    <span
      className="MuiSlider-mark MuiSlider-markActive"
      style={
        Object {
          "left": "0%",
        }
      }
    />
    <span
      className="MuiSlider-markLabel MuiSlider-markLabelActive"
      style={
        Object {
          "left": "0%",
        }
      }
    >
      2006
    </span>
    <span
      className="MuiSlider-mark MuiSlider-markActive"
      style={
        Object {
          "left": "7.142857142857143%",
        }
      }
    />
    <span
      className="MuiSlider-markLabel MuiSlider-markLabelActive"
      style={
        Object {
          "left": "7.142857142857143%",
        }
      }
    />
    <span
      className="MuiSlider-mark MuiSlider-markActive"
      style={
        Object {
          "left": "14.285714285714286%",
        }
      }
    />
    <span
      className="MuiSlider-markLabel MuiSlider-markLabelActive"
      style={
        Object {
          "left": "14.285714285714286%",
        }
      }
    />
    <span
      className="MuiSlider-mark MuiSlider-markActive"
      style={
        Object {
          "left": "21.428571428571427%",
        }
      }
    />
    <span
      className="MuiSlider-markLabel MuiSlider-markLabelActive"
      style={
        Object {
          "left": "21.428571428571427%",
        }
      }
    />
    <span
      className="MuiSlider-mark MuiSlider-markActive"
      style={
        Object {
          "left": "28.571428571428573%",
        }
      }
    />
    <span
      className="MuiSlider-markLabel MuiSlider-markLabelActive"
      style={
        Object {
          "left": "28.571428571428573%",
        }
      }
    />
    <span
      className="MuiSlider-mark MuiSlider-markActive"
      style={
        Object {
          "left": "35.714285714285715%",
        }
      }
    />
    <span
      className="MuiSlider-markLabel MuiSlider-markLabelActive"
      style={
        Object {
          "left": "35.714285714285715%",
        }
      }
    />
    <span
      className="MuiSlider-mark MuiSlider-markActive"
      style={
        Object {
          "left": "42.857142857142854%",
        }
      }
    />
    <span
      className="MuiSlider-markLabel MuiSlider-markLabelActive"
      style={
        Object {
          "left": "42.857142857142854%",
        }
      }
    />
    <span
      className="MuiSlider-mark MuiSlider-markActive"
      style={
        Object {
          "left": "50%",
        }
      }
    />
    <span
      className="MuiSlider-markLabel MuiSlider-markLabelActive"
      style={
        Object {
          "left": "50%",
        }
      }
    >
      2013
    </span>
    <span
      className="MuiSlider-mark MuiSlider-markActive"
      style={
        Object {
          "left": "57.142857142857146%",
        }
      }
    />
    <span
      className="MuiSlider-markLabel MuiSlider-markLabelActive"
      style={
        Object {
          "left": "57.142857142857146%",
        }
      }
    />
    <span
      className="MuiSlider-mark MuiSlider-markActive"
      style={
        Object {
          "left": "64.28571428571429%",
        }
      }
    />
    <span
      className="MuiSlider-markLabel MuiSlider-markLabelActive"
      style={
        Object {
          "left": "64.28571428571429%",
        }
      }
    />
    <span
      className="MuiSlider-mark MuiSlider-markActive"
      style={
        Object {
          "left": "71.42857142857143%",
        }
      }
    />
    <span
      className="MuiSlider-markLabel MuiSlider-markLabelActive"
      style={
        Object {
          "left": "71.42857142857143%",
        }
      }
    />
    <span
      className="MuiSlider-mark MuiSlider-markActive"
      style={
        Object {
          "left": "78.57142857142857%",
        }
      }
    />
    <span
      className="MuiSlider-markLabel MuiSlider-markLabelActive"
      style={
        Object {
          "left": "78.57142857142857%",
        }
      }
    />
    <span
      className="MuiSlider-mark MuiSlider-markActive"
      style={
        Object {
          "left": "85.71428571428571%",
        }
      }
    />
    <span
      className="MuiSlider-markLabel MuiSlider-markLabelActive"
      style={
        Object {
          "left": "85.71428571428571%",
        }
      }
    />
    <span
      className="MuiSlider-mark MuiSlider-markActive"
      style={
        Object {
          "left": "92.85714285714286%",
        }
      }
    />
    <span
      className="MuiSlider-markLabel MuiSlider-markLabelActive"
      style={
        Object {
          "left": "92.85714285714286%",
        }
      }
    />
    <span
      className="MuiSlider-mark MuiSlider-markActive"
      style={
        Object {
          "left": "100%",
        }
      }
    />
    <span
      className="MuiSlider-markLabel MuiSlider-markLabelActive"
      style={
        Object {
          "left": "100%",
        }
      }
    >
      2020
    </span>
    <span
      aria-labelledby="range-slider"
      aria-orientation="horizontal"
      aria-valuemax={2020}
      aria-valuemin={2006}
      aria-valuenow={2006}
      aria-valuetext="2006"
      className="MuiSlider-thumb PrivateValueLabel-thumb-46"
      data-index={0}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      role="slider"
      style={
        Object {
          "left": "0%",
        }
      }
      tabIndex={0}
    >
      <span
        className="PrivateValueLabel-offset-48 MuiSlider-valueLabel"
      >
        <span
          className="PrivateValueLabel-circle-49"
        >
          <span
            className="PrivateValueLabel-label-50"
          >
            2006
          </span>
        </span>
      </span>
    </span>
    <span
      aria-labelledby="range-slider"
      aria-orientation="horizontal"
      aria-valuemax={2020}
      aria-valuemin={2006}
      aria-valuenow={2020}
      aria-valuetext="2020"
      className="MuiSlider-thumb PrivateValueLabel-thumb-46"
      data-index={1}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      role="slider"
      style={
        Object {
          "left": "100%",
        }
      }
      tabIndex={0}
    >
      <span
        className="PrivateValueLabel-offset-48 MuiSlider-valueLabel"
      >
        <span
          className="PrivateValueLabel-circle-49"
        >
          <span
            className="PrivateValueLabel-label-50"
          >
            2020
          </span>
        </span>
      </span>
    </span>
  </span>
</div>
`;
